File: Task_Management_Backend/src/main/java/com/cubeproject/service/TaskService.java
package com.cubeproject.service;

import java.util.List;

import com.cubeproject.model.Task;

public interface TaskService {
	
	 public Task saveTask(Task task);
	 
	 public List<Task> getAllTask(); 
	 
	 public Task getTaskById(int id);
	 
	 public String deleteTask(int id);
	 
	 public Task editTask(Task task, int id);

}

File: Task_Management_Frontend/task_ui/src/service/task.service.js
import axios from "axios";
const API_URL = "http://localhost:8080";

class TaskService{

    saveTask(task) {
        return axios.post(API_URL+"/saveTask",task);
    }

    getAllTask() {
        return axios.get(API_URL + "/");
    }

    getTaskById(id) {
        return axios.get(API_URL + "/" +id);
    }

    deleteTask(id){
        return axios.get(API_URL + "/deleteTask/" +id);
    }

    editTask(task){
        return axios.post(API_URL +"/editTask/" + task.id, task);
    }

}
export default new TaskService;